### DISCUSSION TITLE
Use dependent_fields with checkbox

### DISCUSSION THREAD

**Discussion by CleitonDeLima** [codingjoe/django-select2#282] [Ideas]
If include a field checkbox in `dependent_fields`, the `'on'` value is sent to the view regardless of whether the field is checked or not.

This behavior can be changed [here](https://github.com/codingjoe/django-select2/blob/main/django_select2/static/django_select2/django_select2.js#L30)

```js
$.each(dependentFields, function (i, dependentField) {
  var field = $("[name=" + dependentField + "]", $element.closest("form"));
  if (field.is(":checkbox")) {
    result[dependentField] = field.prop("checked");
  } else {
    result[dependentField] = field.val();
  }
});
```

**Comment by codingjoe** [comment]
Hi @CleitonDeLima,

sure, this sounds like a reasonable enhancement. Would you care to provide a patch?

Cheers!
Joe

**Reply by CleitonDeLima** [reply]
I'll make a patch :+1:

**Reply by codingjoe** [reply]
#289 is ready for you.
